src/test/groovy/graphql/validation/rules/ArgumentsOfCorrectTypeTest.groovy

```diff
@@ -335,6 +335,37 @@ class ArgumentsOfCorrectTypeTest extends Specification {
         validationErrors.get(0).message == "Validation error (WrongType@[dog/doesKnowCommand]) : argument 'dogCommand' with value 'EnumValue{name='PRETTY'}' is not a valid 'DogCommand' - Literal value not in allowable values for enum 'DogCommand' - 'EnumValue{name='PRETTY'}'"
     }
 
+    def "invalid @oneOf argument - has more than 1 key - case #why"() {
+        when:
+        def validationErrors = validate(query)
+
+        then:
+        validationErrors.size() == 1
+        validationErrors.get(0).getValidationErrorType() == ValidationErrorType.WrongType
+        validationErrors.get(0).message == "Validation error (WrongType@[oneOfField]) : Exactly one key must be specified for argument 'oneOfArg' of @oneOf input type 'oneOfInputType'"
```

Comment:
I am not a fan of our validation error messages - and I made them!

One day I wish these where better!